src/app/clusters/scenes-server/SceneTableImpl.h

```diff
@@ -46,89 +46,110 @@ class DefaultSceneHandlerImpl : public scenes::SceneHandler
     DefaultSceneHandlerImpl() = default;
     ~DefaultSceneHandlerImpl() override{};
 
+    /// @brief Encodes an attribute value list into a TLV structure and resizes the buffer to the size of the encoded data
+    /// @param aVlist[in] Attribute value list to encode
+    /// @param serializedBytes[out] Buffer to fill from the Attribute value list in a TLV format
+    /// @return CHIP_ERROR
+    virtual CHIP_ERROR
+    EncodeAttributeValueList(const app::DataModel::List<app::Clusters::Scenes::Structs::AttributeValuePair::Type> & aVlist,
+                             MutableByteSpan & serializedBytes)
+    {
+        TLV::TLVWriter writer;
```

Comment:
Why is the implementation inlined here?  This is virtual anyway...  Please put the implementation in the .cpp?